modules/tensor_mechanics/src/materials/ComputeReturnMappingStress.C

```diff
@@ -39,7 +39,7 @@ ComputeReturnMappingStress::initialSetup()
   std::vector<MaterialName> models = getParam<std::vector<MaterialName> >("return_mapping_models");
   for (unsigned i = 0; i < models.size(); ++i)
   {
-    MooseSharedPointer<RecomputeRadialReturn> rrr = MooseSharedNamespace::dynamic_pointer_cast<RecomputeRadialReturn>(getMaterialSharedPointerByName(models[i]));
+    MooseSharedPointer<StressUpdateBase> rrr = MooseSharedNamespace::dynamic_pointer_cast<StressUpdateBase>(getMaterialSharedPointerByName(models[i]));
```

Comment:
@permcody can you give some input here. Isn't the lifetime management of the material objects the job of the warehouse classes? If so we should probably just use raw pointers here. The design does not require smart pointers, or does it? The classes just get deleted by the warehouse at the end of the simulation.